File: Python/5/5_4.py
def navigate_maze(maze):
    for x in range(len(maze)):
        for y in range(len(maze[x])):
            for z in range(len(maze[x][y])):
                cell = maze[x][y][z]
                if cell.is_wall:
                    print(f"Encountered wall at ({x}, {y}, {z}).")
                elif cell.is_exit:
                    print(f"Exit found at ({x}, {y}, {z}).")
                    return
                else:
                    print(f"Moving through ({x}, {y}, {z}).")
    print("No exit found in the maze.")

File: JavaScript/5/5_4.js
function navigateMaze(maze) {
    for (let x = 0; x < maze.length; x++) {
      for (let y = 0; y < maze[x].length; y++) {
        for (let z = 0; z < maze[x][y].length; z++) {
          let cell = maze[x][y][z];
          handleCell(cell, x, y, z);
        }
      }
    }
    console.log("No exit found in the maze.");
  }
  function handleCell(cell, x, y, z) {
    if (cell.isWall) {
      console.log("Encountered wall at (" + x + ", " + y + ", " + z + ").");
    } else if (cell.isExit) {
      console.log("Exit found at (" + x + ", " + y + ", " + z + ").");
      return true; 
    } else {
      console.log("Moving through (" + x + ", " + y + ", " + z + ").");
    }
    return false;
  }
  
